flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/BridgingFunctionGenUtil.scala

```diff
@@ -288,16 +290,39 @@ object BridgingFunctionGenUtil {
       s"($externalResultTypeTerm) (${typeTerm(externalResultClassBoxed)})"
     }
     val externalResultTerm = ctx.addReusableLocalVariable(externalResultTypeTerm, "externalResult")
+    val externalCode =
+      s"""
+         |${externalOperands.zip(argumentDataTypes)
+            .map{case (element, dataType) => {
+              element match {
+                case external: ExternalGeneratedExpression
+                  if !isInternal(dataType) && !element.resultType.isInstanceOf[StructuredType]
```

Comment:
Why `isInstanceOf[StructuredType]`?
Why not just `dataType  == element.dataType`?